Library/Homebrew/language/node.rb

```diff
@@ -0,0 +1,30 @@
+module Language
+  module Node
+    def self.npm_cache_config
+      "cache=#{HOMEBREW_CACHE}/npm_cache\n"
+    end
+
+    def self.setup_npm_environment
+      # explicitly use our npm and node-gyp executables instead of the user
+      # managed ones in HOMEBREW_PREFIX/lib/node_modules which might be broken
+      ENV.prepend_path "PATH", "#{Formula["node"].opt_libexec}/npm/bin"
+      # explicitly set npm's cache path to HOMEBREW_CACHE/npm_cache to fix
+      # issues under superenv (long build times, high disk space usage)
```

Comment:
This isn't actually a `superenv` thing. We override `HOME` in `formula` which means it happens regardless of env or sandbox status.